src/base/model.js

```diff
@@ -243,16 +243,29 @@ ModelBase.prototype.isNew = function() {
  * @returns {Object} Serialized model as a plain object.
  */
 ModelBase.prototype.serialize = function(options = {}) {
-  const { shallow = false, omitPivot = false, ...rest } = options;
+  const { shallow = false, omitPivot = false, omitNew = false, ...rest } = options;
   const { attributes } = this;
 
+  if (omitNew && this.isNew()) {
+    return null;
+  }
+
```

Comment:
There is no point in performing this verification when `shallow === true`. This shall verified only if `shallow === false`.